Gems/ROS2/Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -48,6 +50,14 @@ namespace ROS2
                 SpawnEntity(service_handle, header, request);
             });
 
+        m_deleteService = ros2Node->create_service<gazebo_msgs::srv::DeleteEntity>(
+            "delete_entity",
+            [this](
+                const DeleteEntityServiceHandle service_handle, const std::shared_ptr<rmw_request_id_t> header, DeleteEntityRequest request)
```

Comment:
What is the reason for using this callback type as opposed to request/response one?

Is it about returning response earlier before Spawn / Despawn entities call is finished?
It would be valid to assume service can be blocked until items are fully spawned / despawned to avoid the need for extra checks on the client side.